crates/rattler_conda_types/src/match_spec/condition.rs

```diff
@@ -0,0 +1,257 @@
+use std::fmt::Display;
+
+use nom::{
+    branch::alt,
+    bytes::complete::tag,
+    character::complete::{char, multispace0, multispace1},
+    sequence::{delimited, preceded},
+    IResult, Parser,
+};
+use serde::Serialize;
+
+use crate::match_spec::parse::matchspec_parser;
+
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct Statement {
+    pub prefix: String,
+    pub condition: Option<MatchSpecCondition>,
+}
+
+/// Represents a condition in a match spec, which can be a match spec itself or a logical combination
+#[derive(Debug, Clone, PartialEq, Serialize, Eq, Hash)]
+pub enum MatchSpecCondition {
+    /// A condition on a certain match spec (e.g. `python >=3.12`)
+    MatchSpec(Box<crate::MatchSpec>),
+    /// A logical AND condition combining two conditions
+    And(Box<MatchSpecCondition>, Box<MatchSpecCondition>),
+    /// A logical OR condition combining two conditions
+    Or(Box<MatchSpecCondition>, Box<MatchSpecCondition>),
+}
+
+impl Display for MatchSpecCondition {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            MatchSpecCondition::MatchSpec(ms) => write!(f, "{ms}"),
+            MatchSpecCondition::And(lhs, rhs) => write!(f, "({lhs} and {rhs})"),
+            MatchSpecCondition::Or(lhs, rhs) => write!(f, "({lhs} or {rhs})"),
+        }
+    }
+}
+
+// Parse whitespace
+fn ws(input: &str) -> IResult<&str, &str> {
+    multispace0(input)
+}
+
+// Parse a matchspec by consuming until we hit a delimiter
+fn matchspec_token(input: &str) -> IResult<&str, &str> {
+    // Try to find the next delimiter
+    let delimiters = ["and", "or", ")", "("];
+
+    // Find the earliest delimiter
+    let mut end_pos = input.len();
+    for delimiter in &delimiters {
+        if let Some(pos) = input.find(delimiter) {
+            // Make sure it's a word boundary for "and"/"or"
+            if *delimiter == "and" || *delimiter == "or" {
+                // Check if it's preceded and followed by whitespace or start/end of string
+                let is_word_boundary = {
+                    let before_ok = pos == 0 || input.chars().nth(pos - 1).unwrap().is_whitespace();
+                    let after_ok = pos + delimiter.len() >= input.len()
+                        || input
+                            .chars()
+                            .nth(pos + delimiter.len())
+                            .unwrap()
```

Comment:
The `unwrap` here basically already checks the precondition that the position is within range. So this could be simplified to `map_or(true, char::is_whitespace)`.